src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CommandResolver.java

```diff
@@ -157,24 +157,24 @@ default void addToMetrics(
     Tag tenantTag = Tag.of(TENANT_TAG, dataApiRequestInfo.getTenantId().orElse(UNKNOWN_VALUE));
     Tags tags = Tags.of(commandTag, tenantTag);
 
-    getIndexUsageTags(logicalExpression, baseIndexUsage);
+    getIndexUsageTags(dbFilterLogicalExpression, baseIndexUsage);
     tags = tags.and(baseIndexUsage.getTags());
 
     meterRegistry.counter(jsonApiMetricsConfig.indexUsageCounterMetrics(), tags).increment();
   }
 
-  private void getIndexUsageTags(LogicalExpression logicalExpression, IndexUsage indexUsage) {
-    for (ComparisonExpression comparisonExpression : logicalExpression.comparisonExpressions) {
-      getIndexUsageTags(comparisonExpression, indexUsage);
+  private void getIndexUsageTags(
```

Comment:
rename , this does not return a value. recordIndexUsage

can merge the function below into this one.

Reply:
solved